lib/ansible/modules/cloud/amazon/rds.py

```diff
@@ -1002,7 +1003,8 @@ def replicate_db_instance(module, conn):
         changed = False
     else:
         try:
-            result = conn.create_db_instance_read_replica(instance_name, source_instance, **params)
+            if not module.check_mode:
+                result = conn.create_db_instance_read_replica(instance_name, source_instance, **params)
```

Comment:
result isn't defined if check_mode but is still being used below in the call to await_resource()